### PR TITLE
samples: Added README for watchdog sample application

### PR DESCRIPTION
**Pull request by Jandro20**
This pull request include a Watchdog README, necessary for issue: https://github.com/zephyrproject-rtos/zephyr/issues/27805.

### CODE REVIEW

**Comment by carlescufi** [comment]
@Jandro20 thanks for the PR! You have some trailing whitespace and you need body text in the commit message

**Comment by Jandro20** [comment]
> @Jandro20 thanks for the PR! You have some trailing whitespace and you need body text in the commit message

Hi @carlescufi, thanks for your review. With these changes, all the merge conflicts will probably be solved.

PS: This is my first contribution to the Zephyr project! But it's not the first time I've used it.

File: samples/drivers/watchdog/README.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using the configuration file in boards/
+
+Building and Running for ST Nucleo F091RC
+=========================================
+
+The sample can be built and executed for the
+:ref:`nucleo_f091rc_board` as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/adc
```
Comment (jeppenodgaard) [inline samples/drivers/watchdog/README.rst:32]: Should be `samples/drivers/watchdog`. Otherwise looks good to me.

**Review by jeppenodgaard — CHANGES_REQUESTED** [review]

File: samples/drivers/watchdog/README.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.. _watchdog sample:
```
Comment (martinjaeger) [inline samples/drivers/watchdog/README.rst:1]:
```suggestion
.. _watchdog-sample:
```

File: samples/drivers/watchdog/README.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Watchdog (WD)
```
Comment (martinjaeger) [inline samples/drivers/watchdog/README.rst:3]: I'm not sure if `WD` is the common abbreviation for watchdogs. The Zephyr APIs use `wdt_` or `WDT_` prefixes. I suggest to leave it out in this title.
```suggestion
Watchdog Sample
```

File: samples/drivers/watchdog/README.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+An useful implementation in our projects is to ensurance that the board is restarted in case of some piece of code
+is kept in an infinite loop.
+
+Depending on the target board, we need to use some configuration included in the /board folder.
+In this sample, we will use the watchdog0 device.
+
+Building and Running
+********************
+
+If the target board allows callbacks, we can implement a way to handle some cases or store some important information.
```
Comment (martinjaeger) [inline samples/drivers/watchdog/README.rst:20]: Please be a bit more specific and rewrite using passive voice instead of "we".

**Review by martinjaeger — CHANGES_REQUESTED** [review]
@Jandro20 Thanks a lot for your first contribution!

Please see my suggested small improvements, mainly related to style and rst markup.

**Comment by Jandro20** [comment]
Hi @martinjaeger, thanks a lot for your review! All the changes you have suggested should be fixed in the last push.

**Review by jeppenodgaard — APPROVED** [review]

**Comment by Jandro20** [comment]
Hey @martinjaeger, can you please review again?

**Review by martinjaeger — APPROVED** [review]
Thanks for the reminder. Looks good to me.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**